File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    ans = 0
    
    while n > 0:
        if n % 2 == 1:  # 홀수면 점프가 필요했던 것
            ans += 1
            n -= 1
        else:           # 짝수면 순간이동으로 온 것 (무료)
            n //= 2
    
    return ans

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;

        while(n>0) {
            // 짝수일 때
            if(n % 2 == 0) {
                n = n / 2;
            }
            else {
                n -= 1;
                ans += 1;
            }
        }

        return ans;
    }
}